app-backend/http4s-util/src/main/scala/com/rasterfoundry/http4s/TracedHTTPRoutes.scala

```diff
@@ -32,34 +32,49 @@ object TracedHTTPRoutes {
   ): Kleisli[OptionT[F, ?], AuthedRequest[F, User], Response[F]] = {
     Kleisli[OptionT[F, ?], AuthedRequest[F, User], Response[F]] { authedReq =>
       val req = authedReq.req
-      val operationName = "http4s-incoming-request"
-      val traceHeader =
+      val operationName = "http4s-request"
+      val tags = Map(
+        "http_method" -> req.method.name,
+        "request_url" -> req.uri.path.toString,
+        "environment" -> Config.environment
+      ).combine {
         req.headers.get(CaseInsensitiveString("X-Amzn-Trace-Id")) match {
-          case Some(header) => header.value
-          case _            => ""
+          case Some(header) => Map("amazon_trace_id" -> header.value)
+          case _            => Map.empty[String, String]
         }
-      val tags = Map(
-        HTTP_METHOD.getKey -> req.method.name,
-        HTTP_URL.getKey -> req.uri.path.toString,
-        "X-Amzn-Trace-Id" -> traceHeader
-      )
+      }
+
+      def transformResponse(context: TracingContext[F]): F[Option[Response[F]]] = {
+        val tracedRequest = AuthedTraceRequest[F](authedReq, context)
+        val responseOptionWithTags = routes.run(tracedRequest) semiflatMap {
+          response =>
+            val traceTags = Map(
+              "http_status" -> response.status.code.toString,
+            ) ++ tags ++
+              response.headers.toList
+                .map(h => (s"response_header_${h.name}" -> h.value))
+                .toMap
+            context
```

Comment:
I _really_ wanted to be able to update `XRayHttp` here ... but it was going to be a little bit of work and it's easy enough to ensure that we get the response code in an annotation (which is mostly what we care about I think and what we'd get). The downside is that we'll have to do a little bit more manual work in the XRay UI.

I think to actually enable updating `http` we'd have to treat it like `Tags` as a `Ref[F, XrayHttp]` and then also create a custom `XRayContext` that can be matched on.